h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -303,8 +303,17 @@ public boolean filter(KeySnapshot.KeyInfo k) {
           // checksum is being calculated: we skip them (see PUBDEV-5286)
           Log.warn("GridSearch encountered concurrent modification while searching DKV", e);
           return false;
-        } catch (RuntimeException e) {
-          if (! (e.getCause() instanceof H2OConcurrentModificationException))
+        } catch (final RuntimeException e) {
+          Throwable ex = e;
+          boolean concurrentModification = false;
+          while (ex.getCause() != null) {
```

Comment:
Do you need to check also suppressed field? 

For example, in theory if we are using try-with-resources and fail in close, then the exception will be listed as suppressed

Reply:
That is a good point... didn't think of it before. In theory, we should.

Reply:
Thinking about it more - this cannot happen in current H2O implementation and handling this properly complicates the code unnecessarily. I would avoid this complexity and leave as-is